lib/experimental/block-bindings-api/html-processing.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Define the mechanism to replpace the HTML depending on the block attributes.
+ *
+ * @package gutenberg
+ */
+
+if ( ! function_exists( 'block_bindings_replace_html' ) ) {
+	/**
+	 * Depending on the block attributes, replace the proper HTML based on the value returned by the source.
+	 *
+	 * @param string $block_content Block Content.
+	 * @param string $block_name The name of the block to process.
+	 * @param string $block_attr The attribute of the block we want to process.
+	 * @param string $source_value The value used to replace the HTML.
+	 */
+	function block_bindings_replace_html( $block_content, $block_name, $block_attr, $source_value ) {
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
+		if ( null === $block_type ) {
+			return;
+		}
+
+		// Depending on the attribute source, the processing will be different.
+		// TODO: Get the type from the block attribute definition and modify/validate the value returned by the source if needed.
+		switch ( $block_type->attributes[ $block_attr ]['source'] ) {
+			case 'html':
+				$p = new WP_HTML_Tag_Processor( $block_content );
+				if ( ! $p->next_tag(
+					array(
+						// TODO: build the query from CSS selector.
+						'tag_name' => $block_type->attributes[ $block_attr ]['selector'],
```

Comment:
Based on #56704, this might not work for the heading block, but you could borrow from @kevin940726's approach in that PR of exploding the selector, at least until the HTML tag processor supports CSS selectors.

Reply:
Thanks for sharing! As part of [this commit](https://github.com/WordPress/gutenberg/pull/56867/commits/5b5546fa9bbd9af29c833b3d4302765fc11a28ae), I'm using that approach for supporting the different selectors we can have for the heading and the button.